x/superfluid/keeper/stake.go

```diff
@@ -141,9 +143,17 @@ func (k Keeper) validateLockForSFDelegate(ctx sdk.Context, lock *lockuptypes.Per
 	if err != nil {
 		return err
 	}
+
+	denom := lock.Coins[0].Denom
+	// If the denom is a concentrated liquidity token, we just strip the position data from the denom
+	if strings.HasPrefix(denom, cltypes.ClTokenPrefix) {
+		index := strings.LastIndex(denom, "/")
+		denom = denom[:index+1]
+	}
+
 	defaultSuperfluidAsset := types.SuperfluidAsset{}
-	if k.GetSuperfluidAsset(ctx, lock.Coins[0].Denom) == defaultSuperfluidAsset {
-		return sdkerrors.Wrapf(types.ErrNonSuperfluidAsset, "denom: %s", lock.Coins[0].Denom)
+	if k.GetSuperfluidAsset(ctx, denom) == defaultSuperfluidAsset {
+		return sdkerrors.Wrapf(types.ErrNonSuperfluidAsset, "denom: %s", denom)
```

Comment:
I don't think this is covered by test

Reply:
It is here https://github.com/osmosis-labs/osmosis/blob/6030a18469e5de91dac4fce3faa4e5dc6c17425a/x/superfluid/keeper/stake_test.go#L188-L198